pyiceberg/catalog/glue.py

```diff
@@ -252,6 +259,22 @@ def _construct_database_input(database_name: str, properties: Properties) -> Dat
     return database_input
 
 
+def _register_glue_catalog_id_with_glue_client(glue: GlueClient, glue_catalog_id: str) -> None:
+    """
+    Register the Glue Catalog ID (AWS Account ID) as a parameter on all Glue client methods.
+
+    It's more ergonomic to do this than to pass the CatalogId as a parameter to every client call since it's an optional
+    parameter and boto3 does not support 'None' values for missing parameters.
+    """
+    event_system = glue.meta.events
+
+    def add_glue_catalog_id(params, **kwargs) -> None:  # type: ignore[no-untyped-def]
```

Comment:
```suggestion
    def add_glue_catalog_id(params: Dict[str, str], **kwargs: Any) -> None:
```
I think it is better to write the type explicitly when possible. WDYT?

Reply:
Sure! I wasn't sure what the type was so I punted, but I will take another pass at this and get the type right.